cpp/src/arrow/compute/kernels/vector_nested.cc

```diff
@@ -107,10 +113,13 @@ struct ListParentIndicesArray {
 
 const FunctionDoc list_flatten_doc(
     "Flatten list values",
-    ("`lists` must have a list-like type.\n"
-     "Return an array with the top list level flattened.\n"
+    ("`lists` must have a logical list type like `[Large]ListType`, \n"
+     "`[Large]ListViewType` and `FixedSizeListType`. \n"
+     "Whether to flatten the top list level or the bottom list level \n"
+     "will be decided based on the `recursive` option specified in \n"
+     ":struct:`ListFlattenOptions`. \n"
```

Comment:
@jorisvandenbossche do you know all the places where this text ends up at? Is it correct? Is the `:struct:` markup going to be interpreted in an useful way?